report/table.go

```diff
@@ -13,54 +13,186 @@ import (
-			label := key[len(template.Prefix):]
-			rows[label] = value
+	})
+
+	// Gather a sorted list of rows' IDs.
+	ids := make([]string, 0, len(rowsByID))
+	for id := range rowsByID {
+		ids = append(ids, id)
+	}
+	sort.Strings(ids)
+
+	// Return the rows in the sorted order.
+	rows = []Row{}
+	for _, id := range ids {
+		rows = append(rows, rowsByID[id])
+	}
+	return rows
+}
+
+// ExtractPropertyList returns the rows to build a property list from this node
+func (node Node) ExtractPropertyList(template TableTemplate) (rows []Row) {
+	valuesByLabel := map[string]string{}
+
+	// Itearate through the whole of our map to extract all the values with the key
+	// with the given prefix as well as the keys corresponding to the fixed table rows.
+	node.Latest.ForEach(func(key string, _ time.Time, value string) {
+		if label, ok := template.FixedRows[key]; ok {
+			valuesByLabel[label] = value
+		} else if label, ok := WithoutPrefix(key, template.Prefix); ok {
+			valuesByLabel[label] = value
 		}
 	})
+
+	// Gather a sorted list of labels.
+	labels := make([]string, 0, len(valuesByLabel))
+	for label := range valuesByLabel {
+		labels = append(labels, label)
+	}
+	sort.Strings(labels)
+
+	// Return the label-value formatted rows sorted by label.
+	rows = []Row{}
+	for _, label := range labels {
+		rows = append(rows, Row{
+			ID: "label_" + label,
+			Entries: map[string]string{
+				"label": label,
+				"value": valuesByLabel[label],
+			},
+		})
+	}
+	return rows
+}
+
+// ExtractTable returns the rows to build either a property list or a generic table from this node
+func (node Node) ExtractTable(template TableTemplate) (rows []Row, truncationCount int) {
+	switch template.Type {
+	case MulticolumnTableType:
+		rows = node.ExtractMulticolumnTable(template)
+	default: // By default assume it's a property list (for backward compatibility).
+		rows = node.ExtractPropertyList(template)
+	}
+
+	truncationCount = 0
 	if str, ok := node.Latest.Lookup(TruncationCountPrefix + template.Prefix); ok {
 		if n, err := fmt.Sscanf(str, "%d", &truncationCount); n != 1 || err != nil {
 			log.Warn("Unexpected truncation count format %q", str)
 		}
 	}
+
 	return rows, truncationCount
 }
 
+// Column is the type for multi-column tables in the UI.
+type Column struct {
+	ID       string `json:"id"`
+	Label    string `json:"label"`
+	DataType string `json:"dataType"`
+}
+
+// Row is the type that holds the table data for the UI. Entries map from column ID to cell value.
+type Row struct {
+	ID      string            `json:"id"`
+	Entries map[string]string `json:"entries"`
+}
+
+// Copy returns a copy of the Row.
+func (r Row) Copy() Row {
+	entriesCopy := make(map[string]string, len(r.Entries))
+	for key, value := range r.Entries {
+		entriesCopy[key] = value
+	}
+	r.Entries = entriesCopy
+	return r
+}
+
 // Table is the type for a table in the UI.
 type Table struct {
-	ID              string        `json:"id"`
```

Comment:
Yup, the only field that has changed here is `Rows` and they only get populated through `ExtractTable` that I kept compatible with the old reports. After that, `Table`s get aggregated up and are indirectly only used in `baseNodeSummary` in `render/detailed/summary.go` which I guess only serves the UI?